### ISSUE TITLE
[Dropdown/Combobox] issues on screen reader (JAWS) in Firefox

### ISSUE THREAD

**Issue by RobertaJHahn** [carbon-design-system/carbon-for-ibm-dotcom#5505]
## Detailed description
This issue has been resolved in Chrome.  This work was managed under issue (#4889 ).  

Copy of the final comment with accessibility tester, Ram:

_Ram has tested the "edit combo" in both chrome and FF.
The behavior you mentioned is found in Chrome.
In FF, the Edit combo role is not identified by jaws. And also the selected value is not announced after making a selection. hence the user would not know the currently selected option.
Regarding the jaws announcement of the input value along with each option in chrome, I think that is tolerable as long as we can achieve the same behavior in FF as in chrome. I would mark that as low sev or even a best practice._

----

Issue:
1. There is no label
2. Screen reader announces this as a list box. Hence the user does not know that this is an Edit combo.
3. When we navigate within the combo box using up/down arrow keys, Screen reader does not announce anything. Hence Unable to select the preferred option 

In React staging, The language selector control is not identified by jaws in arrow key navigation. An issue is already opened on the same and is being tracked.

Expected:
1. Provide appropriate label
2. If this is an editable combo box, Screen reader should convey the "edit combo" role to the user 
3. Screen reader Users should be able to navigate and make the selection within the combo box.
